File: JAVA/2938. Separate Black and White Balls.java
class Solution {
    public long minimumSteps(String s) {
        long res = 0, zero = 0;
        for(int i=s.length()-1;i>=0;--i)
        {
            if(s.charAt(i)=='1')
                res += zero;
            else
                zero++;
        }
        return res;
    }
}

File: C++/2938. Separate Black and White Balls.cpp
class Solution {
public:
    long long minimumSteps(string s) {
        int zero = 0;
        for(auto it:s) if(it=='0') zero++;
        int i = 0;
        int j = 0;
        long long ans = 0;
        int pos = zero;
        while(zero>0 & j<s.size()){
            if(s[j]=='1'){
                ans +=pos-j;
                pos++;
            }
            else{
               zero--;
            }
            j++;
        }
        return ans;
    }
};
